tests/build_tool/test_build_tool.py

```
@@ -19,11 +21,11 @@ def test_success(build_tool_validator, tmp_path, success_test_case):
 
     # exit status
     assert cp.returncode == 0
-
     # logging
-    assert "Remote configuration version: 0.0.1" in cp.stdout
     assert str(success_test_case) in cp.stdout
     assert str(outputdir) in cp.stdout
+    for line in expected_stdout:
```

Do we need to check for completeness? I mean we are checking that each line in in the expected, but not that all the lines of the expected are present. Just asking if that is needed.

---

I'm not sure I understand. The test will pass only if each and every line is in the output. However, should the test fail on this, it will report only one missing string at a time, not all of them. Is this what you mean? We can definitely change it to `assert all(...)`. The semantics would remain the same. Test failures would become more verbose but also more accurate, I suppose.

---

Used ssert all(...) in new commit

---

No, what I mean is that we can have lines in `expected_stdout` that are not present in the actual cp.stdout. I am not sure if we need to check for that.

---

I see. The answer is "no" - for the same reason why you didn't want to test specific log messages.

We want to test only as much of the output that is required to verify that the build tool failed for the expected reason. We don't want the test to depend on the build tool output parts that are not relevant to the test case. The "only as much" part is not an exact definition, so we may need to discuss what is the right amount of expected output on a case by case basis. Does it make sense?

---

Yes, make sense. Just wanted to ask to make sure.